libraries/linux.rb

```diff
@@ -0,0 +1,183 @@
+require 'ffi'
```

Comment:
I think you should wrap this whole file in a `if RUBY_PLATFORM =~ /mswin|mingw|windows/` because `libudev` is not available by default on windows.

Reply:
Done. I forgot that Chef loads all libraries.